src/VolleyManagement.Backend/VolleyManagement.Services/GameService.cs

```diff
@@ -499,62 +499,82 @@ select game
             {
                 if (GameValidation.AreSameTeamsInGames(game, newGame))
                 {
-                    if (GameValidation.IsFreeDayGame(newGame))
-                    {
-                        if (tournamentScheduleInfo.Scheme != TournamentSchemeEnum.PlayOff)
-                        {
-                            throw new ArgumentException(
-                            Resources
-                            .SameFreeDayGameInRound);
-                        }
-                        else
-                        {
-                            throw new ArgumentException(
-                                string.Format(Resources.SameTeamInRound, game.HomeTeamName));
-                        }
-                    }
-                    else
-                    {
-                        throw new ArgumentException(
-                           string.Format(
-                           Resources.SameGameInRound,
-                           game.HomeTeamName,
-                           game.AwayTeamName,
-                           game.Round.ToString()));
-                    }
+                    ValidateAreSameTeamsInGames(game, newGame, tournamentScheduleInfo);
                 }
                 else if (GameValidation.IsTheSameTeamInTwoGames(game, newGame))
                 {
-                    if (GameValidation.IsFreeDayGame(newGame))
-                    {
-                        if (tournamentScheduleInfo.Scheme != TournamentSchemeEnum.PlayOff
-                            && game.HomeTeamId != newGame.HomeTeamId
-                            && game.AwayTeamId != newGame.HomeTeamId)
-                        {
-                            throw new ArgumentException(
-                                Resources.SameFreeDayGameInRound);
-                        }
-                    }
-                    else
-                    {
-                        string oppositeTeam;
+                    ValidateIsTheSameTeamInTwoGames(game, newGame, tournamentScheduleInfo);
+                }
+            }
+        }
 
-                        if (game.HomeTeamId == newGame.HomeTeamId
-                            || game.HomeTeamId == newGame.AwayTeamId)
-                        {
-                            oppositeTeam = game.HomeTeamName;
-                        }
-                        else
-                        {
-                            oppositeTeam = game.AwayTeamName;
-                        }
+        private static void ValidateAreSameTeamsInGames(
+            GameResultDto game, 
+            Game newGame,
+            TournamentScheduleDto tournamentScheduleInfo)
+        {
+            string errorMessage = null;
+            if (GameValidation.IsFreeDayGame(newGame))
+            {
+                if (tournamentScheduleInfo.Scheme != TournamentSchemeEnum.PlayOff)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use the '?:' operator here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3240)